tests/integration/ardrive.int.test.ts

```diff
@@ -0,0 +1,529 @@
+		});
+
+		describe('createPrivateFolder', () => {
+			it('returns the correct ArFSResult', async () => {
+				stub(walletDao, 'walletHasBalance').callsFake(() => {
+					return Promise.resolve(true);
+				});
+				stub(arfsDao, 'getDriveIdForFolderId').callsFake(() => {
+					return Promise.resolve(stubEntityID);
+				});
+				stub(arfsDao, 'getPrivateDrive').callsFake(() => {
+					return Promise.resolve(stubPrivateDrive);
+				});
+				const stubDriveKey = await getStubDriveKey();
+				const result = await arDrive.createPrivateFolder({
+					folderName: 'TEST_FOLDER',
+					driveId: stubEntityID,
+					driveKey: stubDriveKey
+				});
+				assertCreateFolderExpectations(result, 38, urlEncodeHashKey(stubDriveKey));
+			});
+		});
+
+		describe('movePublicFolder', () => {
+			it('returns the correct ArFSResult', async () => {
+				stub(walletDao, 'walletHasBalance').callsFake(() => {
+					return Promise.resolve(true);
+				});
+				stub(arfsDao, 'getDriveIdForFolderId').callsFake(() => {
+					return Promise.resolve(stubEntityID);
+				});
+				stub(arfsDao, 'getPublicDrive').callsFake(() => {
+					return Promise.resolve(stubPublicDrive);
+				});
+				stub(arfsDao, 'getPublicFolder').callsFake(() => {
+					return Promise.resolve(stubPublicFolder);
+				});
+
+				// TODO: SHOULD WE ALLOW MOVE TO SELF?
+				const result = await arDrive.movePublicFolder({
+					folderId: stubEntityID,
+					newParentFolderId: stubEntityID
+				});
+				assertCreateFolderExpectations(result, 20);
+			});
+		});
+
+		describe('movePrivateFolder', () => {
+			it('returns the correct ArFSResult', async () => {
+				stub(walletDao, 'walletHasBalance').callsFake(() => {
+					return Promise.resolve(true);
+				});
+				stub(arfsDao, 'getDriveIdForFolderId').callsFake(() => {
+					return Promise.resolve(stubEntityID);
+				});
+				stub(arfsDao, 'getPrivateDrive').callsFake(() => {
+					return Promise.resolve(stubPrivateDrive);
+				});
+				stub(arfsDao, 'getPrivateFolder').callsFake(() => {
+					return Promise.resolve(stubPrivateFolder);
+				});
+				const stubDriveKey = await getStubDriveKey();
+				// TODO: SHOULD WE ALLOW MOVE TO SELF?
+				const result = await arDrive.movePrivateFolder({
+					folderId: stubEntityID,
+					newParentFolderId: stubEntityID,
+					driveKey: stubDriveKey
+				});
+				assertCreateFolderExpectations(result, 36, urlEncodeHashKey(stubDriveKey));
+			});
+		});
+	});
+
+	describe('file function', () => {
+		describe('uploadPublicFile', () => {
+			it('returns the correct ArFSResult', async () => {
+				stub(walletDao, 'walletHasBalance').callsFake(() => {
+					return Promise.resolve(true);
+				});
+				stub(arfsDao, 'getDriveIdForFolderId').callsFake(() => {
+					return Promise.resolve(stubEntityID);
+				});
+				stub(arfsDao, 'getPublicDrive').callsFake(() => {
+					return Promise.resolve(stubPublicDrive);
+				});
+				stub(arfsDao, 'getPublicFolder').callsFake(() => {
+					return Promise.resolve(stubPublicFolder);
+				});
```

Comment:
We do need to stub the community oracle functions when we upload files. I think we're probably hitting the network here